preview_generator/preview/builder/image__pillow.py

```diff
@@ -59,23 +67,43 @@ def image_to_jpeg_pillow(
                 dims_in=ImgDims(width=image.size[0], height=image.size[1]),
                 dims_out=preview_dims
             )
+            output = BytesIO()
+            image = image.resize((resize_dim.width, resize_dim.height), resample=JPEG_RESAMPLE)
+            if image.mode not in TRANSPARENCY_MODES:
+                # INFO - G.M - 2019-06-18 - Try directly saving new image, this may failed due to
+                # image mode issue.
+                try:
+                    image.save(output, 'jpeg', optimize=JPEG_OPTIMIZE, quality=JPEG_QUALITY, progressive=JPEG_PROGRESSIVE)
+                except OSError as exc:
+                    image = image.convert('RGB')
+                    image.save(output, 'jpeg', optimize=JPEG_OPTIMIZE, quality=JPEG_QUALITY, progressive=JPEG_PROGRESSIVE)
+            else:
+                # INFO - G.M - 2019-06-18 - choose saving mode according to image mode, this should
+                # be jpeg compatible mode
+                if image.mode == 'RGBA':
+                    output_image = Image.new(
+                        'RGB',
+                        (resize_dim.width, resize_dim.height),
+                        (255, 255, 255)
+                    )
+                elif image.mode == 'LA':
+                    output_image = Image.new(
+                        'L',
+                        (resize_dim.width, resize_dim.height),
+                        1
```

Comment:
is this the transparency channel? if so add a comment `# transparency`

Reply:
or better: use a constant

Reply:
code changed in order to be more explicit.